packages/runtime/src/runtime.ts

```diff
@@ -500,6 +547,8 @@ export class Runtime {
   executeTx (txnParams: ExecParams | ExecParams[], program: string,
     args: Uint8Array[]): void {
     const [tx, gtxs] = this.createTxnContext(txnParams); // get current txn and txn group (as encoded obj)
+    // validate first and last rounds
+    this.validateRounds(gtxs);
```

Comment:
```suggestion
    this.validateTxRound(gtxs);
```

Reply:
I have a question. Here we are validating the round for the 'current' transaction only, and i don't think we're incrementing the round after the transaction is executed. Shouldn't we do that? By the current method, if i am executing two or more transactions then the current round (`this.round`) is same for them, which means that those different transactions were executed in the same round.

Reply:
yes, https://algoexplorer.io/
in each round you can have around 18 - 89 transactions committed (observed from the above link).
and it takes 4.5s to commit a block.
therefore it is not clear when to increment rounds(increment after 18 txns or 80 txns) , only user can decide that by setting rounds.

Reply:
Okay, so what does the current round represent in our case? Is it a particular timestamp? I thought we could manage time using rounds and update crowdfunding tests accordingly. Or are these two different?

Reply:
yes, there is a slight difference, here is a good explanation https://developer.algorand.org/docs/features/transactions/#current-round

Reply:
Alright, will check.

Reply:
also, in our case current round is not timestamp, it is value that can be set by user. let's say we want want to check crowdfunding, we will have to set the rounds and then check for validation

Reply:
I have an better idea, let me know if you follow:
+ we will have start timestamp stored - (when calling constructor)
+ at each time when we call a check for transaction, we will have difference ( current time - construction time (in seconds)).
-- round will be `diff / 4.5`  this way we can have good testing for crowdfunding, this is a different idea than what is mentioned in task, what do you think ?

Reply:
yes, got it. I was referring to the commented code we have in crowdfunding tests (related to timestamps, start date, end date..), but logic of rounds is different from that. 
[https://www.pivotaltracker.com/story/show/176631770](https://www.pivotaltracker.com/story/show/176631770)

Reply:
but in this case user cannot set the rounds because round calculation will be changed.

Reply:
I like the idea. 

> round will be diff / 4.5 this way we can have good testing for crowdfunding

I think 4.5s is the average block time for the algorand blockchain. Since we're not on a real blockchain, can't we keep the average block time to be 1s? This way `currentRound` = `lastest timestamp` - `construction timestamp`. 

I will read more about rounds. Because by looking at the difference, let's say transaction is being executed after 2s from the construction time, then `currentRound = 2`. But what does that mean (what does the '2' represent)? 
Also, should we take start timestamp as the a) timestamp when runtime is initialized or b) timestamp when previous tx was executed

Reply:
- current round 2 means 9 seconds are passed i,e two blocks are committed
- if transaction is executed after 2 sec that will be round 1.
- we will take timestamp at time of validation of transaction- let's say you executed transaction after 20 sec after initializing the runtime, so calculation will be done current - init = 20 / 4.5 this will return round.

But i think the task idea is better because we still have to get the rounds from network.
this seems overengineered.

Reply:
there is one more benefit of using task idea, let's say we want to test if application rejects logic after 1 hour, we can set the rounds and check.